tools/generate.py

```diff
@@ -42,20 +42,27 @@ def clang_format(buffer, **kwargs):
 
 
 def api_generate(input_path: Path, output_path: Path):
+    output_path.parent.mkdir(parents=True, exist_ok=True)
     with open(output_path, 'w') as f:
         f.write(clang_format(api.run(input_path)))
 
 
 def te_generate(input_path: Path, output_path: Path):
+    output_path.parent.mkdir(parents=True, exist_ok=True)
     with open(output_path, 'w') as f:
         f.write(clang_format(te.run(input_path)))
 
 
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('-f', '--clang-format', type=Path)
+    parser.add_argument('-D', '--define', type=str, action='append', choices=['have_onnx', 'have_tensorflow'] )
```

Comment:
The flags need to be able to pass to api.py as well so we can run this directly on a single file without having to run generate.py.